src/ui/server/controllers/insurance/your-buyer/index.ts

```diff
@@ -0,0 +1,45 @@
+import { PAGES } from '../../../content-strings';
+import { Request, Response } from '../../../../types';
+import api from '../../../api';
+import { ROUTES, TEMPLATES } from '../../../constants';
+import insuranceCorePageVariables from '../../../helpers/page-variables/core/insurance';
+import { mapCountriesSelect } from '../../../helpers/mappings/map-countries-select';
+import yourBuyerDetailsValidation from './validation/yourBuyerDetailsValidation';
+import { yourBuyerFiledVariables } from '../../../content-strings/fields/insurance/your-buyer';
+
+const getMappedCountries = async (res: Response) => {
+  const countries = await api.external.getCountries();
+  if (!countries || !countries.length) {
+    return res.redirect(ROUTES.PROBLEM_WITH_SERVICE);
+  }
+  return mapCountriesSelect(countries);
+};
+
+export const get = async (req: Request, res: Response) => {
+  const mappedCountries = await getMappedCountries(res);
+  return res.render(TEMPLATES.INSURANCE.YOUR_BUYER.BUYER_BUYER_DETAILS, {
+    ...insuranceCorePageVariables({
+      PAGE_CONTENT_STRINGS: PAGES.INSURANCE.YOUR_BUYER_DETAILS,
+      BACK_LINK: req.headers.referer,
+    }),
+    ...yourBuyerFiledVariables,
+    countries: mappedCountries,
+  });
+};
+
+export const post = async (req: Request, res: Response) => {
+  const mappedCountries = await getMappedCountries(res);
+  const validationErrors = yourBuyerDetailsValidation(req.body);
+  if (validationErrors && Object.keys(validationErrors).length) {
```

Comment:
Very minor can we add a space before `if (validationErrors ...` so it's easier to read, not a blocker